src/components/swap/AdvancedSwapDetails.tsx

```diff
@@ -1,22 +1,42 @@
 import { Trans } from '@lingui/macro'
-import { Percent, Currency, TradeType } from '@uniswap/sdk-core'
+import { Currency, Percent, TradeType } from '@uniswap/sdk-core'
 import { Trade as V2Trade } from '@uniswap/v2-sdk'
 import { Trade as V3Trade } from '@uniswap/v3-sdk'
+import { LoadingRows } from 'components/Loader/styled'
 import { useContext, useMemo } from 'react'
 import { ThemeContext } from 'styled-components/macro'
 import { TYPE } from '../../theme'
 import { computeRealizedLPFeePercent } from '../../utils/prices'
 import { AutoColumn } from '../Column'
 import { RowBetween, RowFixed } from '../Row'
 import FormattedPriceImpact from './FormattedPriceImpact'
-import SwapRoute from './SwapRoute'
+import { TransactionDetailsLabel } from './styleds'
 
 interface AdvancedSwapDetailsProps {
   trade?: V2Trade<Currency, Currency, TradeType> | V3Trade<Currency, Currency, TradeType>
   allowedSlippage: Percent
+  loading?: boolean
 }
 
-export function AdvancedSwapDetails({ trade, allowedSlippage }: AdvancedSwapDetailsProps) {
+function TextWithLoadingPlaceholder({
+  loading,
+  width,
+  children,
+}: {
+  loading: boolean
+  width: number
+  children: React.ReactNode
+}) {
+  return loading ? (
+    <LoadingRows>
+      <div style={{ height: '15px', width: `${width}px` }} />
+    </LoadingRows>
+  ) : (
+    <>{children}</>
```

Comment:
do you need the fragment here?

Reply:
got it to work by changing the type of `children` to `JSX.Element`

Reply:
Might consider `ReactChild | ReactChildren` as another option. Up to you!

Reply:
Hm can't get the typechecked to pass with `ReactChild | ReactChildren`. Might keep `jsx.element`